### ISSUE TITLE
Ignore analyzer rules on generated files

### ISSUE THREAD

**Issue by MattisBrizard** [comigor/artemis#110]
Hi,

Generated files are not following some analyzer rules (ex: public_member_api_docs), would it be possible to add an `//ignore_for_file: ..` with rules ?

Or even better, as possible with json_serializable (coming from source_gen: https://pub.dev/packages/source_gen#configuring-combining_builder-ignore_for_file) a way to specify in bulid.yaml; some rules to ignore in generated files.

```
source_gen|combining_builder:
        options:
          ignore_for_file:
            - public_member_api_docs
```

Thanks :)

**Comment by comigor** [comment]
Hey @MattisBrizard!

Absolutely, that seems important (and thinking about it, has bugged me already in some projects).

Thanks, I'll consider this in the future!

**Comment by vasilich6107** [comment]
It seems to be much easier to exclude them from analyzer
https://github.com/comigor/artemis/pull/101/files

There is a second file generated with json_serializer which could also conflict with your analyzer tool.

**Comment by MattisBrizard** [comment]
Hi,

In theory yes, but this is an old issue on the dart SDK, where the exclude files are still analyzed.
https://github.com/dart-lang/sdk/issues/25551

Concerning the file generated with json_serializable, it takes care of the build.yaml configuration I mentioned above and adds ` ignore_for_file: ...` rules, so the analyzer doesn't take care of the generated files (at least for the rules I configured).

**Comment by vasilich6107** [comment]
Thanks, I did not know about this bug(

**Comment by orestesgaolin** [comment]
It's possible to add ignores for `.g.dart` files by:

```
      source_gen|combining_builder:
        options:
          ignore_for_file:
            - public_member_api_docs
            - lines_longer_than_80_chars
            - constant_identifier_names
```

but it would be great to also have this option for `.graphql.dart` files

**Comment by comigor** [comment]
Released as `6.12.1-beta.1`
